### DISCUSSION TITLE
process.env in nuxt.config not taken into account after deployment

### DISCUSSION THREAD

**Discussion by kaboume** [nuxt/nuxt#21889] [Questions]
Hi,
I'm using the module nuxt-simple-sitemap with this config in my nuxt.config.ts file :

```
export default defineNuxtConfig({
  runtimeConfig: {
    public: {
      API_BASE_URL: '',
    }
  },
 sitemap: {
    dynamicUrlsApiEndpoint: `${process.env.NUXT_PUBLIC_API_BASE_URL}misc/sitemap`,
    inferStaticPagesAsRoutes: false,
  }
...}
```

with this .env file : 

```
PORT=3000
NITRO_PORT=3000
NUXT_PUBLIC_API_BASE_URL=http://localhost:8080/v1/
```

It works well locally but after deployment on mn linux production server, **dynamicUrlsApiEndpoint is undefined**

Here is my pm2 file : 

```
{
    "apps": [
      {
        "name": "website.com",
        "exec_mode": "fork",
        "script": "PORT=3000 NITRO_PORT=3000 NUXT_PUBLIC_API_BASE_URL=https://api.website.com.fr/v1/  node .output/server/index.mjs",
        "cwd": "/var/www/website.com",
        "autorestart": true,
        "watch": false,
        "max_memory_restart": "1G",
        "env": {
          "NODE_ENV": "development"
        },
        "env_production": {
          "NODE_ENV": "production"
        }
      },
```

Note that the API_BASE_URL environment variable is correct in production if I retrieve its contents via useRuntineConfig composable.

Thus, it is only the process.env contained in the nuxt.config file which is not interpreted... How to do it?

**Comment by kaboume** [comment]
No help ?

**Comment by AloisSeckar** [comment]
`nuxt.config.ts` gets serialized during build, therefore you can't have dynamic values in it

Check [here](https://github.com/unjs/nitro/issues/1161#issuecomment-1511444675) for possible workaroud approach using a plugin. Hopefully you will be able to modify it for your case.
